src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -756,21 +773,23 @@ private static int compareImports(String import1, String import2) {
     }
 
     /**
-     * Counts empty lines before given.
-     * @param lineNo
+     * Counts empty lines between given.
+     * @param lineNoFrom
+     *        Line number of previous import.
+     * @param lineNoTo
      *        Line number of current import.
-     * @return count of empty lines before given.
+     * @return count of empty lines between given, exclusive, eg., (lineNoFrom, lineNoTo).
      */
-    private int getCountOfEmptyLinesBefore(int lineNo) {
+    private int getCountOfEmptyLinesBetween(int lineNoFrom, int lineNoTo) {
         int result = 0;
-        final String[] lines = getLines();
-        //  [lineNo - 2] is the number of the previous line
-        //  because the numbering starts from zero.
-        int lineBeforeIndex = lineNo - 2;
-        while (lineBeforeIndex >= 0
-                && CommonUtil.isBlank(lines[lineBeforeIndex])) {
-            lineBeforeIndex--;
-            result++;
+        if (lineNoFrom != lineNoTo) {
+            final String[] lines = Arrays.copyOfRange(getLines(), lineNoFrom, lineNoTo - 1);
```

Comment:
Why do you need to copy `getLines` and not use it directly?

Reply:
I've retested and confirmed that `Arrays.copyOfRange()` pass mutation test easily.

Reply:
But it doesn't answer my original question. it seems like you are copying it for no reason.

Reply:
I'm not sure if this is proper to fake pitest, but the mutation test complains if using `getLines()` and looping with traditional index based `for` loop. The implementation looks the same, Ouch.

Traditional index based `for` loop that mutation test complains:
```java
    private int getCountOfEmptyLinesBetween(int fromLineNo, int toLineNo) {
        int result = 0;
        final String[] lines = getLines();

        for (int i = fromLineNo; i < toLineNo - 1; i++) {
            if (CommonUtil.isBlank(lines[i])) {
                result++;
            }
        }
        return result;
    }
```

Reply:
If pitest fails for these lines, it means you don't have enough cases to fully satisfy them. I consider the copying of the arrays you did more faking out pitest.

What lines are failing pitest and what is error it is giving? You can upload the report to show it.

Reply:
Uploaded pitest report. See https://gmshake.github.io/checkstyle_contribution/pit-reports-pr-7197-traditional-for-loop/201910311151/com.puppycrawl.tools.checkstyle.checks.imports/CustomImportOrderCheck.java.html  line 804

Reply:
https://gmshake.github.io/checkstyle_contribution/pit-reports-pr-7197-traditional-for-loop/201910311151/com.puppycrawl.tools.checkstyle.checks.imports/CustomImportOrderCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@4a8e8f00_804
> 1. changed conditional boundary → SURVIVED

This means `i < toLineNo - 1` was changed to `i <= toLineNo - 1` and it caused no test failures.

If you need help on figuring how to kill this, you can always use regression.
See https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#checkstyle-pitest-regression

You can try running a regression on using `<=` and see if any cases pop up. If none, you can change the condition to something like `i < toLineNo - 2` and so on until pitest is satisfied.

Reply:
@rnveach , is this discussion point resolved?

Reply:
@romani No. `Arrays.copyOfRange` is still used.

Reply:
I'd try as @rnveach suggested.

Reply:
@rnveach  Done at this moment. :)

Reply:
Ping @rnveach

Reply:
@romani it is done